### PR TITLE
Skateboard tweaks and buffs.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
I have been dissatified with the situation of skateboards for a long while now, especially with the 'pro' and 'hoverboard' versions that provide negligeable bonuses for the high price they warrant.

On top of all things, this PR makes the aforementioned two subtypes faster than the standard and improvised skateboards.
Second, the hoverboard now hovers and can be used in zero g (more info in the CL). The thing costs about 2000, so this is pretty much deserved.
Third, skateboards respect move intents. While in walk intent, you will ride it slowly, but you also won't crash against other things.
Fourth, improvised skateboards are a nick more unstable and not mechanically on par with standard skateboards.

## Why It's Good For The Game
Skateboard were cooler back when you would go fast and die young by crashing into an airlock/spessman/wall at stim speed imho, also see the first paragrath of the above section.

## Changelog

:cl:
balance: The pro skateboard and hoverboard are now faster.
balance: The improvised skateboard is a nick more unstable than the standard, so the two aren't exactly the same.
add: Riding a skateboard on walk intent will prevent you from crashing into things, at the cost of speed.
add: hoverboards now actually hovers and can be used even in zero g. There are a caveat to it: It cannot be ridden on open space gaps deeper than one level or actual space, unless there're objects that prevent falls, like lattice or catwalks.
/:cl:

### CODE REVIEW

**Comment by Elli-Skala** [comment]
Does this make hover boards a jetpack?

**Comment by Ghommie** [comment]
> Does this make hover boards a jetpack?

Possibly 🤣 .

**Comment by necromanceranne** [comment]
Does it...let you cross lava safely?

**Comment by Ghommie** [comment]
> Does it...let you cross lava safely? I'm interested in hoverboards for lavaland, that sounds fun.

I'm asking myself that too. I'll give it a test later.

File: code/datums/components/riding/riding_vehicle.dm [hunk 1]
```diff
@@ -169,6 +169,41 @@
 /datum/component/riding/vehicle/scooter/skateboard
 	vehicle_move_delay = 1.5
 	ride_check_flags = RIDER_NEEDS_LEGS | UNBUCKLE_DISABLED_RIDER
+	///If TRUE, the vehicle will be slower (but safer) to ride on walk intent.
+	var/can_slow_down = TRUE
+
+/datum/component/riding/vehicle/scooter/skateboard/RegisterWithParent()
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
```
Comment (MrMelbert) [inline code/datums/components/riding/riding_vehicle.dm:177]:
```suggestion
/datum/component/riding/vehicle/scooter/skateboard/RegisterWithParent()
	. = ..()
	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
```
Also: `RegisterWithParent` but no `UnregisterFromParent`?

**Comment by SimplySmiley123** [comment]
> Does it...let you cross lava safely? I'm interested in hoverboards for lavaland, that sounds fun.

High speed mining... Add the style meter and interactions for it with the sckateboard and it would be a perfect match.

**Comment by Mickyan** [comment]
I never made hoverboards work in space cause proper hoverboards need a surface to hover, they're not flying boards, but that's admittedly a fluff thing

But I'm also not too keen on them being turned it into a jetpack with infinite fuel that doesn't take your backpack slot and that seems dangerously likely to become another thing to powergame while space combat is already a clusterfuck, and I'd really hate to see another thing I made as a fun gimmick be used to kill people better

HOWEVER I think it would actually be really cool if they were allowed to move in zero g but moving came with a stamina cost so they can be used in limited bursts (something about being harder to hold balance with no gravity)

**Comment by Ghommie** [comment]
Ok, I've reworked a bit the zero-g movement of the hoverboard based on your feedback: now it should only work in presence of ground turfs or at most a single z-level above ground turfs, or in the presence of objects that'd prevent falls (e.g. lattice and catwalks).
I've also had to add gravity checks to lava, so that you can actually hover-skate on lava.

**Comment by wesoda25** [comment]
The intent thing is a fantastic bit of QOL, sooo annoying to just be blown off your board bc someone was running the opposite direction.

File: code/datums/components/riding/riding_vehicle.dm [hunk 2]
```diff
@@ -177,8 +179,81 @@
 	set_vehicle_dir_layer(EAST, OBJ_LAYER)
 	set_vehicle_dir_layer(WEST, OBJ_LAYER)
 
+/datum/component/riding/vehicle/scooter/skateboard/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
```
Comment (Jacquerel) [inline code/datums/components/riding/riding_vehicle.dm:184]: Seems like the only thing we add on examine is information about slowing down, but being able to slow down is a boolean.
Is it not going to print that in situations where it is not true?

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request merged**